SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Utils/UserIdContext.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AppCenter.Utils;
+using System;
+
+namespace Microsoft.AppCenter.Windows.Shared.Utils
+{
+    /**
+     * Utility to store and retrieve values for user identifiers.
+    */
+    public partial class UserIdContext
+    {
+        // The lock is static. Instance methods are not necessarily thread safe, but static methods are
```

Comment:
```suggestion
```

That's not the reason we need a static lock. This is for the singleton instance.

Reply:
Done. Not using for UserId field.

Reply:
The comment is not relevant though, should just remove it as per suggestion above.

Reply:
Removed the comment